src/bika/lims/browser/analysisrequest/manage_analyses.py

```diff
@@ -146,11 +146,14 @@ def get_results_range(self):
         result ranges set in analyses. This guarantees that result ranges for
         already present analyses are not overriden after form submission
         """
-        # Extract the result ranges from Sample analyses
-        analyses = self.analyses.values()
-        analyses_rrs = map(lambda an: an.getResultsRange(), analyses)
-        analyses_rrs = filter(None, analyses_rrs)
-        rrs = dicts_to_dict(analyses_rrs, "keyword")
+        # Extract the result ranges from Sample analyses grouped by keyword
+        rrs = {}
+        for analysis in self.analyses.values():
+            results_range = analysis.getResultsRange()
+            if not results_range:
+                continue
+            keyword = analysis.getKeyword()
+            rrs[keyword] = results_range
```

Comment:
Interesting. Any idea why the `keyword` is missing in the case when a Dynamic Analysisspec is used?

Reply:
Haven't looked deeper, but I thought that having the `keyword` in a result range already assigned to an existing analysis was, in fact, meaningless. Therefore, I made the code assume the keyword was not there.